daemon/delete.go

```diff
@@ -128,6 +129,15 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 		return errors.Wrapf(err, "unable to remove filesystem for %s", container.ID)
 	}
 
+	for _, l := range container.HostConfig.Links {
+		_, alias, err := opts.ParseLink(l)
+		if err != nil {
+			return err
+		}
+		fullName := path.Join(container.Name, alias)
+		daemon.releaseName(fullName)
```

Comment:
The actual stored name is in the linkindex, I'd prefer to get what's there vs rebuilding it here.